tools/clang/lib/CodeGen/CGHLSLMSFinishCodeGen.cpp

```diff
@@ -2305,6 +2307,107 @@ bool BuildImmInit(Function *Ctor) {
   return true;
 }
 
+bool IsSingleStoreStaticGlobal(GlobalVariable *GV, DxilTypeSystem &typeSys) {
+  if (GV->getLinkage() != GlobalVariable::LinkageTypes::InternalLinkage)
+    return false;
```

Comment:
User of this function assumes false means there are multiple stores, yet this function returns false in a variety of other cases.  Are all these other cases well understood?  Can they all be reached from HLSL without an internal bug in the compiler?  Are there some cases that represent where we might want to extend support, but are just unsupported for now?  Should we be differentiating internal assumption violation here with DXASSERT?

Reply:
I think just adding an assert before returning false for the null function and missing function annotation cases would be good.

Reply:
Actually missing function annotation could be several intrinsic functions, and determining the special cases here doesn't seem like it would make sense, so leaving it as a return false should do the conservative thing.